clients/openshift.go

```diff
@@ -91,162 +79,6 @@ type FQImage struct {
-}
-
-func convertPolicyRuleToAuthorizationPolicyRule(in *PolicyRule, out *authorization.PolicyRule, s conversion.Scope) error {
-	setDefaultsPolicyRule(in)
-	if err := convertruntimeRawExtensionToruntimeObject(kapi.Scheme, &in.AttributeRestrictions, &out.AttributeRestrictions, s); err != nil {
-		return err
-	}
-
-	out.APIGroups = in.APIGroups
-
-	out.Resources = sets.String{}
-	out.Resources.Insert(in.Resources...)
-
-	out.Verbs = sets.String{}
-	out.Verbs.Insert(in.Verbs...)
-
-	out.ResourceNames = sets.NewString(in.ResourceNames...)
-
-	out.NonResourceURLs = sets.NewString(in.NonResourceURLsSlice...)
-
-	return nil
-
-}
-
-func setDefaultsPolicyRule(obj *PolicyRule) {
-	if obj == nil {
-		return
-	}
-
-	// match the old allow all rule, but only if API groups is nil (not specified in the incoming JSON)
-	oldAllowAllRule := obj.APIGroups == nil &&
-		// avoid calling the very expensive DeepEqual by inlining specific checks
-		len(obj.Verbs) == 1 && obj.Verbs[0] == "*" &&
-		len(obj.Resources) == 1 && obj.Resources[0] == "*" &&
-		len(obj.AttributeRestrictions.Raw) == 0 && len(obj.ResourceNames) == 0 &&
-		len(obj.NonResourceURLsSlice) == 0 &&
-		// semantic equalities will ignore nil vs empty for other fields as a safety
-		// DO NOT REMOVE THIS CHECK unless you replace it with full equality comparisons
-		kapihelper.Semantic.Equalities.DeepEqual(oldAllowAllPolicyRule, *obj)
-
-	if oldAllowAllRule {
-		obj.APIGroups = []string{"*"}
-	}
-
-	// if no groups are found, then we assume ""
-	if len(obj.Resources) > 0 && len(obj.APIGroups) == 0 {
-		obj.APIGroups = []string{""}
-	}
-}
-
-func autoConvertv1PolicyRuleToauthorizationPolicyRule(in *PolicyRule, out *authorization.PolicyRule, s conversion.Scope) error {
-	// WARNING: in.Verbs requires manual conversion: inconvertible types ([]string vs k8s.io/apimachinery/pkg/util/sets.String)
-	if err := runtime.Convert_runtime_RawExtension_To_runtime_Object(&in.AttributeRestrictions, &out.AttributeRestrictions, s); err != nil {
-		return err
-	}
-	out.APIGroups = *(*[]string)(unsafe.Pointer(&in.APIGroups))
-	// WARNING: in.Resources requires manual conversion: inconvertible types ([]string vs k8s.io/apimachinery/pkg/util/sets.String)
-	// WARNING: in.ResourceNames requires manual conversion: inconvertible types ([]string vs k8s.io/apimachinery/pkg/util/sets.String)
-	// WARNING: in.NonResourceURLsSlice requires manual conversion: does not exist in peer-type
-	return nil
-}
-
-func autoConvertV1PolicyRulesToAuthorizationPolicyRules(in []PolicyRule) ([]authorization.PolicyRule, error) {
-	var out []authorization.PolicyRule
-	if in != nil {
-		out = make([]authorization.PolicyRule, len(in))
-		for i := range in {
-			aRule := authorization.PolicyRule{}
-			if err := convertPolicyRuleToAuthorizationPolicyRule(&in[i], &aRule, nil); err != nil {
-				return nil, err
-			}
-			out[i] = aRule
-		}
-	} else {
-		out = nil
-	}
-	return out, nil
-}
-
-/* End of v1 openshift rest calls that are need. */
-
```

Comment:
I guess I can see why you wanted to remove this code. More than I thought there was.